undertow/src/main/java/org/wildfly/extension/undertow/deployment/UndertowDeploymentProcessor.java

```diff
@@ -129,12 +129,43 @@ public void deploy(final DeploymentPhaseContext phaseContext) throws DeploymentU
         if (warMetaData == null) {
             return;
         }
-        String hostName = hostNameOfDeployment(warMetaData, defaultHost);
-        processDeployment(warMetaData, deploymentUnit, phaseContext.getServiceTarget(), hostName);
+        String deploymentName;
+        if (deploymentUnit.getParent() == null) {
+            deploymentName = deploymentUnit.getName();
+        } else {
+            deploymentName = deploymentUnit.getParent().getName() + "." + deploymentUnit.getName();
+        }
+        final UndertowService undertowService = (UndertowService) phaseContext.getServiceRegistry().getService(UndertowService.UNDERTOW).getValue();
+        final Set<Server> servers = undertowService.getServers();
```

Comment:
Turns out we already have hard dependencies in place for all this services (UndertowService, server, host, listener)
only change required was https://github.com/wildfly/wildfly/pull/8422/files#diff-86546f6338488b058859c3c8c5268947L80 which is already part of original PR
currently what we have is:
- host injects(and depends as such) server & UndertowService(root service for subsystem)
- server injects UndertowService (and for default hosts also all its listeners)
- listener injects server

what we have is reverse dependencies, going from lower part of tree (host) up to root of tree (UndertowService) or short host --> server --> undertowService

adding any other dependency in different direction causes CircularDependencyException